File: Dbt.Geomed/Models/Service.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Dbt.Geomed.Models
{
    public class Service : Entity
    {
        public string Name { get; set; }

        [ForeignKey("Category")]
        public long? CategoryId { get; set; }
        public virtual Category Category { get; set; }

        public string ServiceClass { get; set; }
        public string Part { get; set; }
        public string Subpart { get; set; }
        public string Group { get; set; }
        public string Subgroup { get; set; }

        public string GetCode()
        {
            var code = $"{ServiceClass}{Part}.{Subpart}.{Group}";
            if(!String.IsNullOrEmpty(Subgroup))
            {
                code = $"{code}.{Subgroup}";
            }

            return code;
        }
    }
}

File: Dbt.Geomed/ClientApp/src/app/home/models/service.ts
export class Service {
  private _id: number;
  private _name: string;
  private _categoryId: number;

  public constructor(id: number, name: string, categoryId: number) {
    this._id = id;
    this._name = name;
    this._categoryId = categoryId;
  }

  public get id(): number { return this._id; }
  public get name(): string { return this._name; }
  public get categoryId(): number { return this._categoryId; }
}
